test/test-files/sample_daemon_output.ts

```diff
@@ -0,0 +1,562 @@
+    { stream: 'Successfully tagged private-registry-test:1\n' },
+    { status: 'Pulling from library/busybox', id: 'latest' },
+    { status: 'Pulling fs layer',
+    progressDetail: {},
+    id: '90e01955edcd' },
+    { status: 'Downloading',
+    progressDetail: { current: 8214, total: 727978 },
+    progress:
+    '[>                                                  ]  8.214kB/728kB',
+    id: '90e01955edcd' },
+    { status: 'Downloading',
+    progressDetail: { current: 404239, total: 727978 },
+    progress:
+    '[===========================>                       ]  404.2kB/728kB',
+    id: '90e01955edcd' },
+    { status: 'Downloading',
+    progressDetail: { current: 727978, total: 727978 },
+    progress:
+    '[==================================================>]    728kB/728kB',
+    id: '90e01955edcd' },
+    { status: 'Verifying Checksum',
+    progressDetail: {},
+    id: '90e01955edcd' },
+    { status: 'Download complete',
+    progressDetail: {},
+    id: '90e01955edcd' },
+    { status: 'Extracting',
+    progressDetail: { current: 32768, total: 727978 },
+    progress:
+    '[==>                                                ]  32.77kB/728kB',
+    id: '90e01955edcd' },
+    { status: 'Extracting',
+    progressDetail: { current: 727978, total: 727978 },
+    progress:
+    '[==================================================>]    728kB/728kB',
+    id: '90e01955edcd' },
+    { status: 'Pull complete',
+    progressDetail: {},
+    id: '90e01955edcd' },
+    { status:
+    'Digest: sha256:2a03a6059f21e150ae84b0973863609494aad70f0a80eaeb64bddd8d92465812' },
+    { status: 'Status: Downloaded newer image for busybox:latest' },
+    { status: 'The push refers to a repository [127.0.0.1:55055/busybox]' },
+    { status: 'Preparing', progressDetail: {}, id: '8a788232037e' },
+    { status: 'Pushing',
+    progressDetail: { current: 33792, total: 1154353 },
+    progress:
+    '[=>                                                 ]  33.79kB/1.154MB',
+    id: '8a788232037e' },
+    { status: 'Pushing',
+    progressDetail: { current: 1369600, total: 1154353 },
+    progress:
+    '[==================================================>]   1.37MB',
+    id: '8a788232037e' },
+    { status: 'Pushed', progressDetail: {}, id: '8a788232037e' },
+    { status:
+    'latest: digest: sha256:e2d9acbe92a6def141a9f9f2584468206735308df6a696430e25947882385fb2 size: 527' },
+    { progressDetail: {},
+    aux:
+    { Tag: 'latest',
+        Digest:
+        'sha256:e2d9acbe92a6def141a9f9f2584468206735308df6a696430e25947882385fb2',
+        Size: 527 } }
+];
+
+export function* sampleDaemonOutputGenerator(): IterableIterator<string> {
+	for (let step of sampleDaemonOutput) {
+		yield JSON.stringify(step);
+	}
+}
+
+export function* sampleDaemonStreamGenerator(): IterableIterator<string> {
+    for (let step of sampleDaemonOutput) {
+        if ('stream' in step) {
+            yield step.stream!;
```

Comment:
NIT: avoid accessing the attribute twice: 
```typescript
const text = step.stream;
if (text !== undefined) {
    yield text;
}
```
You may need to add typings for `sampleDaemonOutput` above.

Reply:
Hmm, less convinced about this one. :-)  I've changed it to use Array.filter, saving a line of code.